addons/hr_holidays/models/hr_leave.py

```diff
@@ -925,19 +925,19 @@ def action_validate(self):
                 if holiday.leave_type_request_unit != 'day' or any(l.leave_type_request_unit == 'hour' for l in conflicting_leaves):
                     raise ValidationError(_('You can not have 2 leaves that overlaps on the same day.'))
 
+                target_states = {l.id: l.state for l in conflicting_leaves}
                 conflicting_leaves.action_refuse()
                 split_leaves_vals = []
                 for conflicting_leave in conflicting_leaves:
                     if conflicting_leave.leave_type_request_unit == 'half_day' and conflicting_leave.request_unit_half:
                         continue
 
-                    target_state = conflicting_leave.state
```

Comment:
I don't understand what this change fixes, the commit message should explain. Does the state of a leave change as we work through the recordset such that we need to memoise the states before iteration?

Reply:
though looks like this was later changed so much it was removed so this commit should probably also be removed / merged into other stuff.